From e662df2f6db2b90c47cee7d9d78cc80de0c26592 Mon Sep 17 00:00:00 2001
From: Jon TURNEY <jon.turney@dronecode.org.uk>
Date: Mon, 20 Jul 2009 17:10:54 +0100
Subject: [PATCH 5/5] Cygwin/X: Hide native windows for unmapped, non-minimized X windows

Avoid the native windows corresponding to unmapped, non-minimized X windows lingering around

XXX: this isn't quite right, it will miss the mapped -> unmapped transition for a minimized window.
Perhaps we should track the WM_STATE (normal, withdrawn, iconic) of a X window properly.

Signed-off-by: Jon TURNEY <jon.turney@dronecode.org.uk>
---
 hw/xwin/winmultiwindowwindow.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/hw/xwin/winmultiwindowwindow.c b/hw/xwin/winmultiwindowwindow.c
index b05d310..0f87283 100644
--- a/hw/xwin/winmultiwindowwindow.c
+++ b/hw/xwin/winmultiwindowwindow.c
@@ -297,6 +297,10 @@ winUnmapWindowMultiWindow (WindowPtr pWin)
   fResult = (*pScreen->UnrealizeWindow)(pWin);
   WIN_WRAP(UnrealizeWindow, winUnmapWindowMultiWindow);
 
+  /* Hide the unmapped window (unless it is being unmapped because it is minimized) */
+  if (!IsIconic(pWinPriv->hWnd))
+    ShowWindow(pWinPriv->hWnd, SW_HIDE);
+
   return fResult;
 }
 
-- 
1.6.4.2

